app/src/main/java/com/movies/presentation/details/ui/DetailsFragment.kt

```diff
@@ -8,18 +8,28 @@ import com.movies.presentation.base.data.model.MovieUIModel
 import com.movies.presentation.base.fragment.BaseFragment
 import com.movies.presentation.details.ui.ui_state.DetailsUIState
 import com.movies.presentation.details.view_model.DetailsViewModel
+import com.movies.presentation.utils.NavigationConstants.DETAILS
+import com.movies.presentation.utils.NavigationConstants.MOVIE_ID
 import kotlin.reflect.KClass
 
 class DetailsFragment : BaseFragment<DetailsUIState, DetailsViewModel>() {
 
     override val binding by viewBinding(FragmentDetailsBinding::bind)
+
     override val layout: Int get() = R.layout.fragment_details
+
     override val viewModelClass: KClass<DetailsViewModel> get() = DetailsViewModel::class
 
-    private val args by lazy { arguments?.getInt("movieId")!! }
+    override fun showBottomView(): Boolean = false
+
+    override fun resultKey() = DETAILS
 
     override fun onRefresh() = viewModel.fetchMovieDetails(args)
 
+    private val args by lazy { arguments?.getInt(MOVIE_ID)!! }
+
+    private var bundleArg: MovieUIModel? = null
```

Comment:
you dont need bundleArg because you are not using bundle when you getResult

Reply:
I need. With this favourite button is updated when I navigate back from details to home

Reply:
no, you are not using arg on home

Reply:
then why isn't favourites button updated when I delete that line of code?

Reply:
you were right...